src/sota_tools/oauth2.cc

```diff
@@ -24,8 +24,29 @@ size_t curl_handle_write_sstream(void *buffer, size_t size, size_t nmemb, void *
 
 AuthenticationResult OAuth2::Authenticate() {
   CurlEasyWrapper curl_handle;
+  std::string token_suffix = "/token";
+  std::string post_data = "grant_type=client_credentials";
+  auto use_cognito = false;
+  if (server_.length() >= token_suffix.length()) {
+    use_cognito = (0 == server_.compare(server_.length() - token_suffix.length(), token_suffix.length(), token_suffix));
+  }
+
   curlEasySetoptWrapper(curl_handle.get(), CURLOPT_VERBOSE, get_curlopt_verbose());
-  curlEasySetoptWrapper(curl_handle.get(), CURLOPT_URL, (server_ + "/token").c_str());
+  // We need this check for backwards-compatibility with previous versions of treehub.json.
+  // Previous versions have the server URL *without* the token path, so it needs to be hardcoded.
+  // The new version has the full URL with the `/oauth2/token` path at the end, so nothing needs
+  // to be appended. Also we can't send the `scope` to Auth+, it confuses it.
+  // This check can be removed after we finish the migration to Cognito. At that point there's
+  // no need to attempt to be backwards-compatible anymore, since old credentials will have a
+  // client_id that will have been removed from user-profile, and a server URL to Auth+, which
+  // will be in computer heaven.
+  // check similar implementation in Scala for garage-sigh here:
```

Comment:
```suggestion
  // check similar implementation in Scala for garage-sign here:
```
Sounds good in general, though!